src/validation.cpp

```diff
@@ -4996,10 +5033,25 @@ bool LoadBlockIndex(const CChainParams& chainparams)
         pblocktree->WriteFlag("txindex", fTxIndex);
         LogPrintf("%s: transaction index %s\n", __func__, fTxIndex ? "enabled" : "disabled");
 
-        // Use the provided setting for -assetindex in the new database
-        fAssetIndex = gArgs.GetBoolArg("-assetindex", DEFAULT_ASSETINDEX);
-        pblocktree->WriteFlag("assetindex", fAssetIndex);
-        LogPrintf("%s: asset index %s\n", __func__, fAssetIndex ? "enabled" : "disabled");
+        // Use the provided setting for -rewards in the new database
+        fRewardsEnabled = gArgs.GetBoolArg("-rewards", DEFAULT_REWARDS_ENABLED);
+        pblocktree->WriteFlag("rewards", fRewardsEnabled);
+        LogPrintf("%s: rewards %s\n", __func__, fRewardsEnabled ? "enabled" : "disabled");
+
+        // Rewards requires txindex and assetindex
+        if (fRewardsEnabled) {
+            LogPrintf("%s: rewards has enabled the asset index\n", __func__);
+
+            fAssetIndex = true;
+            pblocktree->WriteFlag("assetindex", fAssetIndex);
+            LogPrintf("%s: asset index %s\n", __func__, fAssetIndex ? "enabled" : "disabled");
+        }
+        else {
+            // Use the provided setting for -assetindex in the new database
+            fAssetIndex = gArgs.GetBoolArg("-assetindex", DEFAULT_ASSETINDEX);
+            pblocktree->WriteFlag("assetindex", fAssetIndex);
+            LogPrintf("%s: asset index %s\n", __func__, fAssetIndex ? "enabled" : "disabled");
+        }
```

Comment:
Just want to double check. 
So if rewards is enabled, it enables assetIndex.

Reply:
The rewards enablement flag has been removed.